dusty/http_server.py

```diff
@@ -25,6 +25,15 @@ def _app_name_from_forwarding_info(hostname, port):
                 return app.name
     raise ValueError('Could not find app for {}:{}'.format(hostname, port))
 
+@app.route('/shutdown', methods=['POST'])
+def shutdown():
+    logging.info('Received POST request to shut down HTTP server')
+    shutdown_function = request.environ.get('werkzeug.server.shutdown')
```

Comment:
This is a dark incantation from Flask, I don't know why Armin thought this was more straightforward than just providing an importable `shutdown` function but here we are